ext/opentelemetry-ext-jaeger/src/opentelemetry/ext/jaeger/__init__.py

```diff
@@ -0,0 +1,369 @@
+        tag = _convert_attribute_to_tag(attribute_key, attribute_value)
+        if tag is None:
+            continue
+        tags.append(tag)
+    return tags
+
+
+def _convert_attribute_to_tag(key, attr):
+    """Convert the attributes to jaeger tags."""
+    if isinstance(attr, bool):
+        return jaeger.Tag(key=key, vBool=attr, vType=jaeger.TagType.BOOL)
+    if isinstance(attr, str):
+        return jaeger.Tag(key=key, vStr=attr, vType=jaeger.TagType.STRING)
+    if isinstance(attr, int):
+        return jaeger.Tag(key=key, vLong=attr, vType=jaeger.TagType.LONG)
+    if isinstance(attr, float):
+        return jaeger.Tag(key=key, vDouble=attr, vType=jaeger.TagType.DOUBLE)
+    logger.warning("Could not serialize attribute %s:%r to tag", key, attr)
+    return None
+
+
+class AgentClientUDP:
+    """Implement a UDP client to agent.
+
+    Args:
+        host_name: The host name of the Jaeger server.
+        port: The port of the Jaeger server.
+        max_packet_size: (Optional) Maximum size of UDP packet.
+        client: Class for creating new client objects for agencies. It should
+            extend from the agent :class: `.AgentIface` type and implement
+            :meth:`.AgentIface.emitBatch`. Default and only option to
+            :class:`.AgentClient`.
+    """
+
+    def __init__(
+        self,
+        host_name: str,
+        port: int,
+        max_packet_size: int = UDP_PACKET_MAX_LENGTH,
+        client=agent.Client,
+    ):
+        self.address = (host_name, port)
+        self.max_packet_size = max_packet_size
+        self.buffer = TTransport.TMemoryBuffer()
+        self.client = client(
+            iprot=TCompactProtocol.TCompactProtocol(trans=self.buffer)
+        )
+
+    def emit(self, batch: jaeger.Batch):
+        """
+        Args:
+            batch: Object to emit Jaeger spans.
+        """
+
+        self.client._seqid = 0
+        #  truncate and reset the position of BytesIO object
+        self.buffer._buffer.truncate(0)
+        self.buffer._buffer.seek(0)
+        self.client.emitBatch(batch)
+        buff = self.buffer.getvalue()
+        if len(buff) > self.max_packet_size:
+            logger.warning(
+                "Data exceeds the max UDP packet size; size %r, max %r",
+                len(buff),
+                self.max_packet_size,
+            )
+            return
+
+        with socket.socket(socket.AF_INET, socket.SOCK_DGRAM) as udp_socket:
+            udp_socket.sendto(buff, self.address)
+
+
+class Collector:
+    """Submits collected spans to Thrift HTTP server.
+
+    Args:
+        thrift_url: URL of the Jaeger HTTP Thrift.
+        auth: (Optional) Auth tuple that contains username and password for
+            Basic Auth.
+        http_transport: Class for creating new client for Thrift HTTP server.
+    """
+
+    def __init__(
+        self,
+        thrift_url: str = "",
+        auth: typing.Tuple[str, str] = None,
```

Comment:
Doesn't this need to be `Optional[Tuple[str, str]]` to match the default `None` value? `mypy` doesn't seem to complain about it, but that may be because it's masked by other problems.